addons/utm/models/ir_http.py

```diff
@@ -16,7 +16,7 @@ def _set_utm(cls, response):
         domain = cls.get_utm_domain_cookies()
         for url_parameter, __, cookie_name in request.env['utm.mixin'].tracking_fields():
             if url_parameter in request.params and request.httprequest.cookies.get(cookie_name) != request.params[url_parameter]:
-                response.set_cookie(cookie_name, request.params[url_parameter], domain=domain)
+                response.set_cookie(cookie_name, request.params[url_parameter], domain=domain, cookie_type='optional')
```

Comment:
These are not just optional cookies, they are tracking cookies. And most of the optional cookies are not flagged e.g. totp device cookies and frontend lang cookies are not technically required for odoo to *work* (although the totp device cookie has its own opt-in so probably counts as required once the user asks for it). Possibly whatever crm_reveal does as well, I've no idea.

Reply:
Note that most of the cookies have been checked by FP and PDE which decided, after long debates, at least for the website related ones, if they were required or not.

While we still don't really agree about some like the website lang, that's as per the spec.
There have been long discussion about this already, but if you feel like this should absolutely be changed, I guess we can yet again discuss that with them.
But AFAIC, we always have to fight to make cookies bar related stuff like this PR move on, so any step further is a win tbh, whether the website lang has to be marked as required or not.

About the crm_reveal one, he is flagged as optional as it should IMO.

About the totp one, that's the only one we are unsure if it should be marked as optional. In doubt, we left it as required. We don't want to mess with this one so close to the release.

Let us know if you think something is to be changed.